src/insight/systemInsight.go

```diff
@@ -0,0 +1,181 @@
+package insight
+
+import (
+	"errors"
+	"time"
+
+	"github.com/accuknox/auto-policy-discovery/src/libs"
+	ipb "github.com/accuknox/auto-policy-discovery/src/protobuf/v1/insight"
+	sys "github.com/accuknox/auto-policy-discovery/src/systempolicy"
+	types "github.com/accuknox/auto-policy-discovery/src/types"
+)
+
+func ConvertSysInsDataToInsResponse(sysdata ipb.SystemInsightData) ipb.InsightResponse {
+	var locsysinsdata ipb.SystemInsightData
+	var insresp ipb.InsightResponse
+
+	locsysinsdata.ContainerName = sysdata.ContainerName
+	locsysinsdata.SysResource = sysdata.SysResource
+
+	insresp.ClusterName = sysdata.ClusterName
+	insresp.NameSpace = sysdata.Namespace
+	insresp.Labels = sysdata.Labels
+	insresp.SystemResource = append(insresp.SystemResource, &locsysinsdata)
+
+	return ipb.InsightResponse{
+		ClusterName:    insresp.ClusterName,
+		NameSpace:      insresp.NameSpace,
+		Labels:         insresp.Labels,
+		SystemResource: insresp.SystemResource,
+	}
+}
+
+func convertWPFSToInsightData(wpfsSet types.ResourceSetMap) types.SysInsightResponseData {
+	var resData types.SysInsightResponseData
+
+	for wpfs, fsset := range wpfsSet {
+		var locFsData types.SystemData
+		var locObsData types.SysInsightData
+
+		// Populate Fileset data(fromsource, process paths and file paths)
+		locFsData.FromSource = wpfs.FromSource
+		if wpfs.SetType == sys.SYS_OP_FILE {
+			locFsData.FilePaths = append(locFsData.FilePaths, fsset...)
+		}
+		if wpfs.SetType == sys.SYS_OP_PROCESS {
+			locFsData.ProcessPaths = append(locFsData.ProcessPaths, fsset...)
+		}
+		if wpfs.SetType == sys.SYS_OP_NETWORK {
+			locFsData.NetworkPaths = append(locFsData.NetworkPaths, fsset...)
+		}
+
+		if len(resData.SysData) > 0 {
+			idx := 0
+			for _, locResData := range resData.SysData {
+				if locResData.ClusterName == wpfs.ClusterName && locResData.Namespace == wpfs.Namespace &&
+					locResData.Labels == wpfs.Labels && locResData.ContainerName == wpfs.ContainerName {
+					resData.SysData[idx].SysProcessFileData = append(resData.SysData[idx].SysProcessFileData, locFsData)
+					break
+				}
+				idx++
+			}
+
+			if idx == len(resData.SysData) {
+				locObsData.ClusterName = wpfs.ClusterName
+				locObsData.Namespace = wpfs.Namespace
+				locObsData.Labels = wpfs.Labels
+				locObsData.ContainerName = wpfs.ContainerName
+				locObsData.SysProcessFileData = append(locObsData.SysProcessFileData, locFsData)
+
+				resData.SysData = append(resData.SysData, locObsData)
+			}
+		} else {
+			locObsData.ClusterName = wpfs.ClusterName
+			locObsData.Namespace = wpfs.Namespace
+			locObsData.Labels = wpfs.Labels
+			locObsData.ContainerName = wpfs.ContainerName
+			locObsData.SysProcessFileData = append(locObsData.SysProcessFileData, locFsData)
+
+			resData.SysData = append(resData.SysData, locObsData)
+		}
```

Comment:
duplication

Reply:
I will check if it is possible to ignore if else and bring in one if case.

Reply:
This duplication cannot be ignored.
If case in non-zero case and else case runs only once when index = 0. 

I will check the possibility to bring all under one loop.